src/models/application.ts

```diff
@@ -455,40 +451,51 @@ const getApplicationModel = function (
 		 * 	console.log(device);
 		 * })
 		 */
-		async getWithDeviceServiceDetails(
-			slugOrUuidOrId: string | number,
-			options?: PineOptions<Application>,
-		): Promise<
-			Application & {
-				owns__device: Array<DeviceWithServiceDetails<CurrentServiceWithCommit>>;
-			}
-		> {
-			options ??= {};
-			const serviceOptions = mergePineOptions(
-				{
-					$expand: [
-						{
-							owns__device: {
-								$expand: getCurrentServiceDetailsPineExpand(true),
-							},
-						},
-					],
+		async getWithDeviceServiceDetails<
+			T extends ODataOptionsWithoutCount<Application['Read']>,
+		>(slugOrUuidOrId: string | number, options?: T) {
+			const baseOptions = {
+				$expand: {
+					owns__device: {
+						$expand: getCurrentServiceDetailsPineExpand,
+					},
 				},
-				options,
-			);
+			} as const;
+
+			const serviceOptions = mergePineOptions(baseOptions, options);
 
 			const app = (await exports.get(
 				slugOrUuidOrId,
 				serviceOptions,
-			)) as Application & {
-				owns__device: Array<DeviceWithServiceDetails<CurrentServiceWithCommit>>;
-			};
+			)) as unknown as NonNullable<
+				OptionsToResponse<
+					Application['Read'],
+					typeof baseOptions,
+					typeof slugOrUuidOrId
+				>
+			>;
+
 			if (app.owns__device) {
```

Comment:
btw I think that this `if` was just to make TS happy, so feel free to remove it.